src/main/java/com/criticalsoftware/AnnouncementResource.java

```diff
@@ -0,0 +1,84 @@
+package com.criticalsoftware;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.core.Response;
+import org.bson.types.ObjectId;
+
+@Path("/announcements")
+public class AnnouncementResource {
+
+    @Inject
+    private AnnouncementRepository repository;
+
+    @DELETE
+    @Path("/{id}")
+    public Response delete(@PathParam("id") String id) {
+        try {
+            // Validate ObjectId
+            if (!ObjectId.isValid(id)) {
+                return Response.status(Response.Status.BAD_REQUEST).entity("Invalid announcement ID").build();
+            }
+
+            ObjectId objectId = new ObjectId(id);
+            Announcement announcement = repository.findById(objectId);
+            if (announcement == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("Announcement not found.").build();
+            }
+
+            // Use the repository method to delete by ID
+            repository.deleteById(id);
+
+            return Response.noContent().build();
+        } catch (Exception e) {
+            // Log exception for debugging
+            e.printStackTrace();
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("An error occurred while deleting the announcement").build();
+        }
+    }
+
+    // Method to update an announcement by its ID
+    @PUT
+    @Path("/{id}")
+    public Response update(@PathParam("id") String id, Product product) {
+        try {
+            // Validate if the provided ID is a valid ObjectId
+            if (!ObjectId.isValid(id)) {
+                return Response.status(Response.Status.BAD_REQUEST).entity("Invalid announcement ID").build();
+            }
+
+            // Convert the string ID to ObjectId
+            ObjectId objectId = new ObjectId(id);
+
```

Comment:
This is well done, but I want you to try not to user PRoduct as a parameter, but a request instead.
See Martim's PR and we will talk about this.